rebuild/test/unit/PublicKey.test.ts

```diff
@@ -0,0 +1,93 @@
+          "pkBytes is 32 bytes, but must be 48 or 96 bytes long"
+        );
+      });
+      it("should take uncompressed byte arrays", () => {
+        expectEqualHex(
+          PublicKey.deserialize(validPublicKey.uncompressed).serialize(false),
+          validPublicKey.uncompressed
+        );
+      });
+      it("should take compressed byte arrays", () => {
+        expectEqualHex(PublicKey.deserialize(validPublicKey.compressed), validPublicKey.compressed);
+      });
+      it("should create jacobian or affine points", () => {
+        expectEqualHex(PublicKey.deserialize(validPublicKey.compressed, CoordType.affine), validPublicKey.compressed);
+        expectEqualHex(PublicKey.deserialize(validPublicKey.compressed, CoordType.jacobian), validPublicKey.compressed);
+      });
+      describe("argument validation", () => {
+        for (const [type, invalid] of invalidInputs) {
+          it(`should throw on invalid pkBytes type: ${type}`, () => {
+            expect(() => PublicKey.deserialize(invalid)).to.throw("pkBytes must be a BlstBuffer");
+          });
+        }
+        it("should throw incorrect length pkBytes", () => {
+          expect(() => PublicKey.deserialize(Buffer.alloc(12, "*"))).to.throw(
+            "pkBytes is 12 bytes, but must be 48 or 96 bytes long"
+          );
+        });
+      });
+      it("should throw on invalid key", () => {
+        expect(() => PublicKey.deserialize(sullyUint8Array(validPublicKey.compressed))).to.throw("BLST_BAD_ENCODING");
+      });
+    });
+  });
+  describe("methods", () => {
+    describe("serialize", () => {
+      const sk = SecretKey.deserialize(SECRET_KEY_BYTES);
+      const pk = sk.toPublicKey();
+      it("should default to compressed serialization", () => {
+        expectEqualHex(pk, pk.serialize(true));
+        expectNotEqualHex(pk, pk.serialize(false));
+      });
+      it("should serialize compressed to the correct length", () => {
+        expect(pk.serialize(true)).to.have.lengthOf(BLST_CONSTANTS.PUBLIC_KEY_LENGTH_COMPRESSED);
+      });
+      it("should serialize uncompressed to the correct length", () => {
+        expect(pk.serialize(false)).to.have.lengthOf(BLST_CONSTANTS.PUBLIC_KEY_LENGTH_UNCOMPRESSED);
+      });
+      it("should serialize affine and jacobian points to the same value", () => {
+        const jacobian = PublicKey.deserialize(pk.serialize(), CoordType.jacobian);
+        const affine = PublicKey.deserialize(pk.serialize(), CoordType.affine);
+        expectEqualHex(jacobian.serialize(true), affine.serialize(true));
+        expectEqualHex(jacobian.serialize(false), affine.serialize(false));
+      });
+    });
+    describe("keyValidate()", () => {
+      it("should not throw on valid public key", () => {
+        const pk = PublicKey.deserialize(validPublicKey.uncompressed);
+        expect(pk.keyValidate()).to.be.undefined;
+      });
+      // TODO: I cannot create an invalid public key to test with
```

Comment:
was there a previous example of invalid public key?

Reply:
there is a `badPublicKey` in the fixtures

Reply:
the `badPublicKey`, or any invalid key for that matter, will no deserialize in order to run the function.  its not possible to pass a bad key into P1 or P1_Affine.  The blst library throws.